File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/routines/VoidProcedure.scala
package dev.typr.foundations.docs.routines
import dev.typr.foundationssc.*

import java.sql.SQLException

@SuppressWarnings(Array("unused"))
object VoidProcedure:
  private val tx: Transactor = null // placeholder

  //start
  // A void procedure — no OUT parameters, just side effects
  val auditLog: DbProcedure.Def2_0[String, String] =
    DbProcedure.define("audit_log")
      .in(PgTypes.text)       // action
      .in(PgTypes.text)       // details
      .build()

  @throws[SQLException]
  def logAction(action: String, details: String): Unit =
    auditLog.call(action, details).transact(tx)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/routines/VoidProcedure.java
package dev.typr.foundations.docs.routines;

import dev.typr.foundations.*;

@SuppressWarnings("unused")
public class VoidProcedure {
  Transactor tx = null; // placeholder

  // start
  // A void procedure — no OUT parameters, just side effects
  static final DbProcedure.Def2_0<String, String> auditLog =
      DbProcedure.define("audit_log").input(PgTypes.text).input(PgTypes.text).build();

  void logAction(String action, String details) {
    auditLog.call(action, details).transact(tx);
  }
  // stop
}
